lib/client.js

```diff
@@ -90,6 +122,25 @@ exports.createClient = function(reference) {
       exchangePrefix:   reference.exchangePrefix || ''
     }, _defaultOptions);
 
+    // Shortcut for which default agent to use...
+    var isHttps = this._options.baseUrl.indexOf('https') === 0;
+
+    if (this._options.agent) {
+      // We have explicit options for new agent create one...
+      this._httpAgent = isHttps ?
```

Comment:
We could this is what other tools do but it is PITA to set it up correctly which I why I chose this instead. A simple instanceof check would work if you want both?